src/harm.cpp

```diff
@@ -78,6 +79,73 @@ void Harm::AddChild(Object *child)
     Modify();
 }
 
+TransPitch Harm::GetRootPitch(unsigned int &pos)
+{
+    Text *textObject = dynamic_cast<Text *>(this->GetChild(0, TEXT));
+    assert(textObject);
+    std::wstring text = textObject->GetText();
+
+    if (text.at(pos) >= 'A' && text.at(pos) <= 'G') {
```

Comment:
This will crash if `text` is an empty. Also, since you are retrieving the Text child in each method, it would be better to simply get it in `Harm::Transpose`. Then check that 1) it is not NULL, and 2) not empty. Then you can pass the `text` variable as parameter to the four methods, without actually setting it back in them but only at the end of `Harm::Transpose`. This means without calling `SetText` each time. This would make the four methods static-like methods, but I think it would be fine to leave them as they are. Alternatively, they could be moved to the Text class. This would be more correct in terms of classes encapsulation. However, having the transpose related methods in Harm makes it somehow clearer. I am fine either way.

Reply:
I changed it to check text.length() before any accesses. I did not refactor the textObject retrieval.